tests/trainer/test_tp.py

```diff
@@ -1,27 +1,456 @@
 # Copyright 2022 MosaicML Composer authors
 # SPDX-License-Identifier: Apache-2.0
 
+from typing import Any, Optional, Sequence, TypeVar, Union
+
+E = TypeVar('E', bound=BaseException)
+
+import numpy as np
 import pytest
 import torch
 from packaging import version
-from torch.utils.data import DataLoader
+from torch.distributed._tensor import DTensor, Replicate
+from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+from torch.utils.data import DataLoader, Dataset
 
+from composer.callbacks import MemoryMonitor
+from composer.loggers import InMemoryLogger
 from composer.trainer.trainer import Trainer
-from composer.utils import dist
+from composer.utils import FSDPConfig, ParallelismConfig, TPConfig, dist, reproducibility
 from tests.common import (
     RandomClassificationDataset,
+    SimpleComposerMLP,
     SimpleModel,
     world_size,
 )
 
 
+class RandomClassificationDatasetReplicated(Dataset):
```

Comment:
should this be in datasets?

Reply:
can this be a subclass for simplicity?

Reply:
i don't think subclassing would make things simpler as we'd have to overwrite every single method of `RandomClassificationDataset`

Reply:
* moved to datasets
* I subclassed at the end